cli/local/delete.go

```diff
@@ -18,30 +18,62 @@ package local
 
 import (
 	"fmt"
+	"strings"
 
+	"github.com/cortexlabs/cortex/pkg/consts"
 	"github.com/cortexlabs/cortex/pkg/lib/docker"
 	"github.com/cortexlabs/cortex/pkg/lib/errors"
+	"github.com/cortexlabs/cortex/pkg/lib/prompt"
 	"github.com/cortexlabs/cortex/pkg/operator/schema"
 	"github.com/cortexlabs/cortex/pkg/types/spec"
 )
 
-func Delete(apiName string, keepCache bool) (schema.DeleteResponse, error) {
+func Delete(apiName string, keepCache, deleteForce bool) (schema.DeleteResponse, error) {
 	_, err := docker.GetDockerClient()
 	if err != nil {
 		return schema.DeleteResponse{}, err
 	}
 
 	var apiSpec *spec.API = nil
-	if !keepCache {
-		if apiSpec, err = FindAPISpec(apiName); err != nil {
+	if apiSpec, err = FindAPISpec(apiName); err != nil {
```

Comment:
It looks like this error isn't being handled if `errors.GetKind(err) != ErrCortexVersionMismatch` and `keepCache == false`.

Reply:
The reason why the error isn't handled is that `apiSpec` is only required for `DeleteCachedModels` function as `apiSpec.ModelIDS()`.

You could say that `DeleteAPI` also has to call `FindAPISpec` too, but then, if we were to just return the error, then the containers wouldn't get deleted, which is a part of what `DeleteAPI` does. I'd say the thinking would be like this: _"don't care whether FindAPISpec fails when the cache is to be kept because I'll do the DeleteAPI call which does the proper clean-up anyway and if it errors, then I'll get the correct error (wrapped and all that)"_.

We could add `DeleteAPI` and then just return the error in that `if` branch. As it is right now, that would have the same effect. What do you think?